src/dra/6D59C.c

```diff
@@ -0,0 +1,871 @@
+#define INCLUDE_ASM_NEW
+#include "dra.h"
+#include "sfx.h"
+#include "player.h"
+
+u8 c_D_800ACF18[] = {10, 8, 8, 6, 6, 4, 4,   4,   4, 4,
+                     4,  4, 4, 4, 4, 4, 255, 255, 0, 0};
+
+void func_8010D59C(void) {
+    byte stackpad[40];
+    Primitive* prim;
+    s32 i;
+    s32 tempAC;
+
+    if (g_Entities[1].ext.ent1.unk0 != 0) {
+        return;
+    }
+    // This is disgusting and should be able to do && and || but I couldn't make
+    // it work. Compiler wanted to optimize with -2 and SLTIU and such. If you
+    // can get this to not be ridiculous you will be my hero. For now, at least
+    // this compiles right.
+    tempAC = PLAYER.ext.player.unkAC;
+    if (tempAC >= 0x5D) {
+        if (tempAC < 0x5F) {
+            g_Entities[1].ext.ent1.unk2 = 10;
+            return;
+        }
+        // This should really be a simple && but that fails to match.
+        if (0x63 > tempAC) {
+            if (tempAC >= 0x60) {
+                g_Entities[1].ext.ent1.unk2 = 10;
+                return;
+            }
+        }
+    }
+    if ((g_Player.padTapped & ~(PAD_START | PAD_SELECT)) ||
```

Comment:
I made GAMEBUTTONS. I agree that this doesn't seem related to the reset combo. I think it's just checking if the player has pressed any button that affects the game state (since start and select will pause the game state, but won't actually cause any action). I am happy to change GAMEBUTTONS to something else. I'm confused about how it uses padTapped, padHeld, and padPressed - this kind of seems like someone didn't know how to properly use the pad variables to test for something...